system/http.go

```diff
@@ -209,3 +217,12 @@ func (u *DefHTTP) Body() (io.Reader, error) {
 
 	return u.resp.Body, nil
 }
+
+func hasUserAgentHeader(headers []string) bool {
+	for _, header := range headers {
+		if strings.HasPrefix(header, USER_AGENT_HEADER_PREFIX) {
```

Comment:
Do we need to lowercase the input to guarantee match?

Reply:
Yes we do need that :)